sacc/covariance.py

```diff
@@ -464,3 +488,39 @@ def inverted(self):
             Inverse covariance
         """
         return np.diag(1.0/self.diag)
+
+    def get_dense(self):
+        """Return a copy of the covariance matrix in dense form.
+
+        Returns
+        --------
+
+        C: array
+            The full covariance
+        """
+        return np.diag(self.diag)
+
+
+def concatenate_covariances(*covariances):
+    # If all the covariances are diagonal then the concatenated
+    # version can be diagonal
+    if all(isinstance(cov, DiagonalCovariance) for cov in covariances):
+        variances = np.concatenate([cov.diag for cov in covariances])
+        return DiagonalCovariance(variances)
+
+    # Otherwise we have to get things in a common form, and
+    # make a block-diagonal covariance.
+    blocks = []
+
+    # For each of the pieces we extract any blocks
+    # that will go into the concatenation
+    for cov in covariances:
+        # For an existing block-diagonal covariance
+        # we retain the block structure
+        if isinstance(cov, BlockDiagonalCovariance):
+            blocks += cov.blocks
+        # For everything else we just use a dense matrix
+        else:
+            blocks.append(cov.get_dense())
+
+    return BlockDiagonalCovariance(blocks)
```

Comment:
There is not a newline at the end of this file.

Reply:
Fixed